lib/Backend/Lower.cpp

```diff
@@ -5591,6 +5591,13 @@ Lowerer::InsertOneLoopProbe(IR::Instr *insertInstr, IR::LabelInstr *loopLabel)
     IR::RegOpnd *regStackPointer = IR::RegOpnd::New(
         NULL, this->m_lowererMD.GetRegStackPointer(), TyMachReg, this->m_func);
 
+#if _M_ARM64
+    // On ARM64 RegSP can only be used in specific contexts, and CMP is not one of them. Instead it has to be moved to a temp register as an ADD.
+    IR::RegOpnd* tmp = regStackPointer;
+    regStackPointer = IR::RegOpnd::New(NULL, SCRATCH_REG, TyMachReg, this->m_func);
```

Comment:
>regStackPointer [](start = 4, length = 15)

Change to 

IR::RegOpnd* scratchReg = IR::RegOpnd::New(NULL, SCRATCH_REG, TyMachReg, this->m_func);